ikarus/utils/functionhelper.hh

```diff
@@ -39,6 +43,43 @@ void obtainLagrangeNodePositions(const LV& localView,
   for (auto& nCoord : lagrangeNodeCoords)
     nCoord = localView.element().geometry().global(nCoord);
 }
+
+/**
+ * \brief A helper function to obtain the global index from the global positions for a Lagrange node
+ * \ingroup utils
+ * \tparam size Size of the nodal coordinate vector
+ * \tparam FEC Type of the finite element container.
+ * \tparam CI Type of the child index defining the direction to be fixed
+ *
+ * \param fes Finite element container.
+ * \param pos Global position
+ * \param childIndex Index of the child in a power basis defining the direction
```

Comment:
I think it would be better if you get the global position for each direction at the same time. Otherwise if you want to fix each direction of a node, you need to use the function multiple times for all child indices